library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -157,43 +187,25 @@ private boolean hasValidFileLocation(Migration.FileMetadata metadata) {
 
     private void migrateCompleteDownloads(InternalMigrationStatus migrationStatus,
                                           SqlDatabaseWrapper database,
-                                          MigrationExtractor migrationExtractor,
+                                          List<Migration> completeMigrations,
                                           DownloadsPersistence downloadsPersistence,
                                           String basePath) {
-        List<Migration> migrations = migrationExtractor.extractMigrations();
-        Log.d(TAG, "migrations are all EXTRACTED, time is " + System.nanoTime());
-
-        migrationStatus.markAsMigrating();
-        onUpdate(migrationStatus);
-        Log.d(TAG, "about to migrate the files, time is " + System.nanoTime());
-
-        for (int i = 0, size = migrations.size(); i < size; i++) {
-            migrationStatus.update(i, size);
-            onUpdate(migrationStatus);
-
-            Migration migration = migrations.get(i);
+        Log.d(TAG, "Complete migrations starting " + System.nanoTime());
+        for (Migration completeMigration : completeMigrations) {
             downloadsPersistence.startTransaction();
             database.startTransaction();
 
-            migrateV1DataToV2Database(downloadsPersistence, migration, basePath, true);
-            deleteFrom(database, migration);
+            migrateV1DataToV2Database(downloadsPersistence, completeMigration, basePath, true);
+            deleteFrom(database, completeMigration);
 
             downloadsPersistence.transactionSuccess();
             downloadsPersistence.endTransaction();
             database.setTransactionSuccessful();
             database.endTransaction();
+            migrationStatus.migrationComplete();
+            onUpdate(migrationStatus);
         }
-
-        Log.d(TAG, "all data migrations are COMMITTED, about to delete the old database, time is " + System.nanoTime());
-
-        migrationStatus.markAsDeleting();
```

Comment:
Not relevant to `complete migrations` moved to it's own method that executes after **all** migrations are completed.